File: app/src/main/java/facebook/com/socialrunner/domain/data/repository/RunnerRepository.java
package facebook.com.socialrunner.domain.data.repository;

import com.google.firebase.database.DatabaseReference;

import facebook.com.socialrunner.domain.data.entity.Runner;

public class RunnerRepository extends FirebaseRepository<Runner> {

    private static final String RUNNER_ENTITY_PATH = "runners";

    public RunnerRepository() {
        super(RUNNER_ENTITY_PATH);
    }

    public void fetchByName(String name, ResultHandler<Runner> handler) {
        fetchByPath(RUNNER_ENTITY_PATH + "/" + name, handler);
    }
}

File: app/src/main/java/facebook/com/socialrunner/domain/data/repository/RunnerRepository.kt
package facebook.com.socialrunner.domain.data.repository


import facebook.com.socialrunner.domain.data.entity.Entity
import facebook.com.socialrunner.domain.data.entity.Position
import facebook.com.socialrunner.domain.data.entity.Runner

class RunnerRepository : FirebaseRepository<Entity>(RUNNER_ENTITY_PATH) {

    fun fetchByName(name: String, handler: ResultHandler<Runner>) {
        fetchByField("name", name, handler)
    }

    fun fetchLocation(username: String, handler: ResultHandler<String>) {
        fetchByName(username, ResultHandler { runner ->
            runner?.let {
                fetchByPath(getPathWithId(runner.id!!) + "/position", handler)
            }
        })
    }

    fun updateLocation(runnerId: String, pos: Position) {
        db.getReference(getPathWithId(runnerId) + "/position").setValue(pos)
    }

    companion object {
        private const val RUNNER_ENTITY_PATH = "runners"
    }
}
